# Capacity Comparison: 100-Person Space Station vs. ISS
## Known aspects of the ISS (or Data)
The volume of the ISS is:
- Total Pressurized Volume, $(V_{tot})_{ISS}$: $916 \,m^3$
- Habitable Volume, $(V_{hab})_{ISS}$: $388 \,m^3$

This means about $42\%$ of the ISS's total volume is habitable. It typically hosts a:
- Standard crew, $n_{avg} = 7$ astronauts.
- Maximum crew of,$n_{max} = 13$ astronauts (short-term).

Not sure when this will matter in my analysis but the ISS mass is $420,000 \, kg$.

In [1]:
ISS_Pressurized_Volume = 916 #m^3
ISS_Habitable_Volume = 388 #m^3
ISS_min_crew = 6 #astronauts
ISS_max_crew = 13 #astronauts
max_volume_per_crew_ISS = ISS_Habitable_Volume/ISS_min_crew
min_volume_per_crew_ISS = ISS_Habitable_Volume/ISS_max_crew
ideal_volume_per_crew_ISS = ISS_Pressurized_Volume/ISS_min_crew
print("max volume per crew", max_volume_per_crew_ISS)
print("min volume per crew", min_volume_per_crew_ISS)
print("ideal volume per crew", ideal_volume_per_crew_ISS)
print("habitable percentage", 100*ISS_Habitable_Volume/ISS_Pressurized_Volume)

max volume per crew 64.66666666666667
min volume per crew 29.846153846153847
ideal volume per crew 152.66666666666666
habitable percentage 42.35807860262009


## Known aspects of Von Braun's Rotating Wheel Space Station
Wernher von Braun's [video](https://www.youtube.com/watch?v=5JJL8CUfF-o) says that a rotating wheel space station would have 9 sections for:
1. Headquarter and Communications
2. Earth Weather Observation/Prediction
3. Military Reconnaisance using Optical and Radar Telescopes
4. Emergency Hospital Section
5. Astronomy
6. Calculation Machines and a lower/outer deck with maintenance facilities
7. A/C equipment
8. Living quarters
9. Botanical and Zoological Laboratory
The video shows two decks throughout the space station, which I assume is 6 $m$ height in total. Wikipedia says 3 decks, but I will stick to two 6 $m$ decks for now.

The system is powered by an atomic reactor attached on one side of its central hub; its heat will drive a turbogenerator to power the station.
Radio and radar antennas sit atop the reactor. The other side of the hub has an airlock for astronauts to access/exit the space station via smaller spacecraft. Like a bicycle wheel, the system had thin spokes are condensers for the turbogenerator and air-conditioning plant. Three larger tubes are elevator shafts that connect the hub to the main habitable sections of the wheel station. The station would:
- host 50 astronauts though [Wikipedia](https://en.wikipedia.org/wiki/Rotating_wheel_space_station#History) states 80 people.
- rotate at roughly 3 rpm to produce artificial gravity on the rim; and
- be a torus of 250 $ft$ (76  $m$) diameter. The distance from the center of the tube to the center of the torus is $R = 70/2 = 35 \, m$ and the minor radius of the tube is $r = 6/2 = 3 \, m$. The volume of this torus is then the station's total pressurized volume, which is given by:
$(V_{tot})_{vB} = 2 \pi^2 R r^2$

So, we can estimate the volume of the the von Braun space station as:
- Total Pressurized Volume: 6217.85 $m^3$
- Habitable Volume: 3730.71 $m^3$

In [2]:
from math import pi
vb_wheel_dia = 70 #m
vb_wheel_height = 6 #m

vb_wheel_Pressurized_Volume = 2 * pi**2 * (vb_wheel_dia/2) * (vb_wheel_height/2)**2 #m^3
habitable_factor_ISS = ISS_Habitable_Volume/ISS_Pressurized_Volume
habitable_factor_targeted = 0.6
vb_wheel_Habitable_Volume = vb_wheel_Pressurized_Volume * habitable_factor_targeted  #m^3

vb_wheel_std_crew = 50 #astronauts
vb_wheel_max_crew = 100 #astronauts
max_volume_per_crew_vb = vb_wheel_Habitable_Volume/vb_wheel_std_crew
min_volume_per_crew_vb = vb_wheel_Habitable_Volume/vb_wheel_max_crew
ideal_volume_per_crew_vb = vb_wheel_Pressurized_Volume/vb_wheel_std_crew
print("vb_wheel_Pressurized_Volume", vb_wheel_Pressurized_Volume)
print("vb_wheel_Habitable_Volume", vb_wheel_Habitable_Volume)
print("max volume per crew", max_volume_per_crew_vb)
print("min volume per crew", min_volume_per_crew_vb)
print("ideal volume per crew", ideal_volume_per_crew_vb)

vb_wheel_Pressurized_Volume 6217.850772686295
vb_wheel_Habitable_Volume 3730.710463611777
max volume per crew 74.61420927223554
min volume per crew 37.30710463611777
ideal volume per crew 124.35701545372591


In [3]:
habitable_ratios = vb_wheel_Habitable_Volume/ISS_Habitable_Volume
Pressurized_ratios = vb_wheel_Pressurized_Volume/ISS_Pressurized_Volume
print("habitable_ratios of von Braun's wheel vs ISS", habitable_ratios)
print("Pressurized_ratios of von Braun's wheel vs ISS", Pressurized_ratios)

habitable_ratios 9.6152331536386
Pressurized_ratios 6.788046695072375


### How did von Braun imagine its realisation?
Pre-fabricated and tested on the ground after which it is dismantled and assembled robotically in-orbit.

First, the hub is inflated using compressed air from built-in tanks. Protection from meteors is ensured by putting thin metal plates; this is unnecessary today as we have developed high-strength textiles that are used on inflatable space stations.

A second launch brings with it an airlock, which is again attached by robots; this allows the pressurized hub to be used as a temporary quarters as the station assembly continues with astronaut-robot teams.

A number successive launches (he says) ensure the "parts of the station  are assembled in the correct order". Nylon tethers prevent parts from drifting away.

The reactor appears to be the next big piece that is installed so I assume this is the third launch. This is followed by the elevator shafts and the rim sections being brought together. We can assume a single launch for each module: $3$ elevator tubes and 9 rim modules (assuming one per section).

![](../assets/imgs/AssemblyOfRim.png)

Assuming condenser pipes are installed from one launch (but the strucutre itself needs to be unwound robotically, which seems unreasonable/unreal at the moment as a robotic task.
![](../assets/imgs/CondenserPipesInstallation.png)

Lastly, the installation of instruments/equipment (a simple catch-all term for imaging antennas and the lab facilities compute for astornomical/EO data analysis, I think) and two small rocket motors on the rim. These rocket motors "blast for a few seconds" to revolve the station at about $3 \, RPM$. 



### Some Percieved Gaps in Design:
1. Assumes a launch every 24 hours- not close to it then and not really close to that kind of cadence today. What could we do today?
2. Doesn't really say how many launches are needed to realize assembly in the video. This can probably be found elsewhere but my estimate from watching the video is 15-18 launches.
3. The process involves a number of assembly robots but we barely have one of these today. And whatever is there not autonomous.

### Question: How can we double the total volume of the von Braun wheel?
To double the wheel's pressurized volume to 12435.7 $m^3$, one option is to increase the tube's height to 8.48 $m$. 

In [4]:
from math import sqrt
vb_wheel_height_for_doubled_volume = 6 * sqrt(2) #m
vb_wheel_Pressurized_Volume_doubled = 2 * pi**2 * (vb_wheel_dia/2) * (vb_wheel_height_for_doubled_volume/2)**2 #m^3
print("doubled vb_wheel_Pressurized_Volume", vb_wheel_Pressurized_Volume_doubled, "when tube dia is", vb_wheel_height_for_doubled_volume)

doubled vb_wheel_Pressurized_Volume 12435.701545372594 when tube dia is 8.485281374238571


The other option is to increase the torus's diameter altogether, which we discuss later.

### Question: Are there reasons to increase the radius of the wheel as opposed to the height of the tube?
One benefit might be that you need to rotate the tube slower to generate artificial gravity.

In [5]:
vb_wheel_radius = vb_wheel_dia / 2
earth_g = 9.81 #m/s^2
rot_speed_vb = sqrt(1/3*earth_g/vb_wheel_radius)
print("rot speed of von Braun wheel to generate a third of Earth g is", rot_speed_vb, "rad/s or", rot_speed_vb/0.10472, "RPM")

rot speed of von Braun wheel to generate a third of Earth g is 0.30566087650952556 rad/s or 2.918839538861016 RPM


Comparing my calculations to [2001: A Space Odyssey's Space Staiton V](https://youtu.be/im-JM0f_J7s?t=136)

In [6]:
SSV_wheel_radius = 150 #m
moon_g = 1.655 #m/s^2
rot_speed_SSV_for_moon_g = sqrt(moon_g/SSV_wheel_radius)
print("2001's wheel generates Moon's gravity by rotating at", rot_speed_SSV_for_moon_g, "rad/s or", rot_speed_SSV_for_moon_g/0.10472, "RPM")
rot_speed_SSV_for_earth_g = sqrt(1/3*earth_g/SSV_wheel_radius)
print("2001's wheel generates third of Earth's gravity by rotating at", rot_speed_SSV_for_earth_g, "rad/s or", rot_speed_SSV_for_earth_g/0.10472, "RPM")
rot_speed_SSV_for_earth_g = sqrt(earth_g/SSV_wheel_radius)
print("2001's wheel generates third of Earth's gravity by rotating at", rot_speed_SSV_for_earth_g, "rad/s or", rot_speed_SSV_for_earth_g/0.10472, "RPM")

2001's wheel generates Moon's gravity by rotating at 0.10503967504392488 rad/s or 1.003052664666968 RPM
2001's wheel generates third of Earth's gravity by rotating at 0.147648230602334 rad/s or 1.4099334473102942 RPM
2001's wheel generates third of Earth's gravity by rotating at 0.25573423705088844 rad/s or 2.442076366032166 RPM


This shows that lower spin rates (needed to prevent motion sickness and acceleration differentials between that push blood to the feet) result in accelerations closer to Earth's but are achievable only with structures larger in diameter than the von Braun wheel. But building larger structures carries the benefit of more pressurized volumes; Space Station V is a little over 4 times the volume of the von Braun wheel and 29 times the ISS.

In [7]:
SSV_Pressurized_Volume = 2 * pi**2 * (SSV_wheel_radius) * (vb_wheel_height/2)**2 #m^3
print("SSV Pressurized volume is", SSV_Pressurized_Volume)
print("SSV pressurized volume is", SSV_Pressurized_Volume/vb_wheel_Pressurized_Volume, "time the von Braun wheel's and", SSV_Pressurized_Volume/ISS_Pressurized_Volume, "times that of ISS.")

SSV Pressurized volume is 26647.931882941266
SSV pressurized volume is 4.285714285714286 time the von Braun wheel's and 29.09162869316732 times that of ISS.


SSV volume is 4.285714285714286 time the von Braun wheel's and 29.09162869316732 times that of ISS.

## So how can we get to this in one launch- that's the question!
Are inflatables the answer....?

In [12]:
# Calculate the diameter of a torus with an internal volume of 100,000 m^3 and inner tube diameter of 10 m
from math import pi
# Given values
V_hph = 100000  # m^3, internal volume
d_hph = 10  # m, inner tube diameter (small diameter)

# Calculate the radius of the inner tube
r_hph = d_hph / 2

# The volume of a torus is given by the formula: V = 2π^2 * R * r^2
# Where R is the distance from the center of the torus to the center of the tube (large radius)

# Rearrange the formula to solve for R
R_hph = V_hph / (2 * pi**2 * r_hph**2)

# Calculate the overall diameter of the torus
D_hph = 2 * (R_hph + r_hph)

print(f"For a torus with an internal volume of {V_hph} m^3 and an inner tube diameter of {d_hph} m:")
print(f"The large radius (R) is approximately {R_hph:.2f} m")
print(f"The overall diameter of the torus is approximately {D_hph:.2f} m")

# Compare to Space Station V
print(f"\nComparison to ISS:")
print(f"But it has a volume of {V_hph/ISS_Pressurized_Volume:.2f} times that of ISS")

print(f"\nComparison to von Braun's wheel:")
print(f"But it has a volume of {V_hph/vb_wheel_Pressurized_Volume:.2f} times that of von Braun's wheel")



For a torus with an internal volume of 100000 m^3 and an inner tube diameter of 6 m:
The large radius (R) is approximately 562.90 m
The overall diameter of the torus is approximately 1131.79 m

Comparison to ISS:
But it has a volume of 109.17 times that of ISS

Comparison to von Braun's wheel:
But it has a volume of 16.08 times that of von Braun's wheel


In [13]:
R_hph = 100  # m^3, internal volume
d_hph = 10  # m, inner tube diameter (small diameter)

# Rearrange the formula to solve for R
V_hph = 2 * pi**2 * R_hph * r_hph**2
# Calculate the overall diameter of the torus
D_hph = 2 * (R_hph + r_hph)

print(f"For a torus with an internal volume of {V_hph} m^3 and an inner tube diameter of {d_hph} m:")
print(f"The large radius (R) is approximately {R_hph:.2f} m")
print(f"The overall diameter of the torus is approximately {D_hph:.2f} m")

# Compare to Space Station V
print(f"\nComparison to ISS:")
print(f"But it has a volume of {V_hph/ISS_Pressurized_Volume:.2f} times that of ISS")

print(f"\nComparison to von Braun's wheel:")
print(f"But it has a volume of {V_hph/vb_wheel_Pressurized_Volume:.2f} times that of von Braun's wheel")

For a torus with an internal volume of 17765.287921960844 m^3 and an inner tube diameter of 10 m:
The large radius (R) is approximately 100.00 m
The overall diameter of the torus is approximately 206.00 m

Comparison to ISS:
But it has a volume of 19.39 times that of ISS

Comparison to von Braun's wheel:
But it has a volume of 2.86 times that of von Braun's wheel
